builder/hcloud/step_create_server.go

```diff
@@ -116,41 +122,45 @@ func (s *stepCreateServer) Cleanup(state multistep.StateBag) {
 	}
 }
 
-func setRescue(ctx context.Context, client *hcloud.Client, server *hcloud.Server, rescue string, sshKeys []*hcloud.SSHKey) error {
+func setRescue(ctx context.Context, client *hcloud.Client, server *hcloud.Server, rescue string, sshKeys []*hcloud.SSHKey) (string, error) {
 	rescueChanged := false
```

Comment:
```suggestion
	rescueChanged := false
	rescuePassword:= ""
```

Reply:
We don't need the value `rescuePassword`, we can directly return the Password when it is needed.